maven-scm-providers/maven-scm-providers-git/maven-scm-provider-jgit/src/main/java/org/apache/maven/scm/provider/git/jgit/command/info/JGitInfoCommand.java

```diff
@@ -74,6 +76,8 @@ protected ScmResult executeCommand(
             return new InfoScmResult(infoItems, new ScmResult("JGit.resolve(HEAD)", "", objectId.toString(), true));
         } catch (Exception e) {
             throw new ScmException("JGit resolve failure!", e);
+        } finally {
+            JGitUtils.closeRepo(git);
```

Comment:
This shouldn't make a difference, no?

Reply:
It does! `git.close()` does not close the underlying repo unfortunately.

Reply:
Insane, this is unlogical.